### PR TITLE
refactor: move handlers to lib/handlers

### LINKED ISSUE #1257: MQTT keepalive less than 60s and chrome 88 intensive timer throttling
- it would be great to have some documentation regarding best practices in regards to keepalive and chrome intensive timer throttling
- it is not uncommon for our application tab to be put in the background by our users since they can also spend a fair amount of time in other web applications
- our keepalive is less than 60s since we need to know fairly quickly when the connection with the broker has been severed since we require timely message delivery
- chome 88  only executes timers every minute during intensive timer throttling which means our keepalive does not happen within the configured time frame which in turn causes disconnects
- please advise, thanks

https://docs.google.com/document/d/105Hlbcb1mTV06bT9aDpwBotY13eZiKV5Hl4PW5LlyeE/edit
https://developer.chrome.com/blog/timer-throttling-in-chrome-88/

AB#9481421

### PR DESCRIPTION
**Pull request by robertsLando**
- Created `client.log` function that by default uses `debug` but can be overriden using `options.log`
- Created `client.nop`
- Moved all `client._handle` in `lib/handlers`
- Create `MqttClient.defaultId` static method
- Ensure shiftPingInterval is called every time a packet is written to the stream. Could fix #1257. An alternative could also to shift the ping interval every time a pingresp is received from server

### CODE REVIEW

File: test/client_mqtt5.js [hunk 1]
```diff
@@ -617,16 +617,16 @@ describe('MQTT 5.0', function () {
...
+    const client = mqtt.connect(opts)
+    server.once('client', function (c) {
       console.log('server received client')
-      serverClient.on('auth', function (packet) {
+      c.on('auth', function (packet) {
         console.log('serverClient received auth: packet %o', packet)
-        serverClient.end(done)
+        client.end(done)
       })
     })
```
Comment (robertsLando) [inline test/client_mqtt5.js:628]: @BertKleewein By checking all failed actions seems the only test that occasionally keeps failing was this one. Let's see if this change fixes that too

**Review by robertsLando — COMMENTED** [review]
Check

File: test/client_mqtt5.js (hunk 1 above)
Comment (robertsLando) [reply test/client_mqtt5.js:628]: Nope that doesn't fix the test neighter... I have a feel this is caused by a race condition, the auth packet could be sent even before the connack is received and the `c.on('auth')` listener may be set after the auth is received causing the test to fail

**Comment by robertsLando** [comment]
@vishnureddy17 @BertKleewein Could you tell me your thoughts about this?

Once this is merged I will continue with https://github.com/mqttjs/MQTT.js/issues/1637. Standard doesn't highlight errors like eslint, like undefined vars and others, that's absolutely bad and difficult to see at first sight expecially when refactoring things...

**Review by vishnureddy17 — APPROVED** [review]
Looks great, much needed refactor.

**Comment by robertsLando** [comment]
@BertKleewein My idea right now is to make the code cleaner and easier to work with without breaking to much stuff. I understand what you mean and I think we can reach that goal, anyway mocking the client to unit test those handlers wouldn't be that hard right now, but I understand you would like to isolate the access those handlers have, that's possible too but we can keep that for another PR :) 

My next steps will be to remove standard with eslint+prettier (sure this will highlight lot of other bugs in code) and then thinking about moving to Typescript. For the second part what scares me most is rewriting tests... Converting all the files to typescript could require me a day but the tests are a lot. Anyone that could help me with that when I start?

**Pull request merged**